src/entities/DealTokenSwap.ts

```diff
@@ -259,6 +259,48 @@ export class DealTokenSwap implements IDeal {
     return this.dealDocument.votingSummary.primaryDAO.votes.concat(this.dealDocument.votingSummary.partnerDAO?.votes ?? []);
   }
 
+  @computedFrom("dealDocument.votingSummary.primaryDAO.votes")
+  public get primaryDAOVotes(): Array<IVoteInfo> {
+    return this.dealDocument.votingSummary.primaryDAO.votes;
+  }
+
+  @computedFrom("dealDocument.votingSummary.partnerDAO.votes")
+  public get partnerDAOVotes(): Array<IVoteInfo> {
+    return this.dealDocument.votingSummary.partnerDAO?.votes ?? [];
+  }
+
+  @computedFrom("allVotes")
+  public get submittedVotes(): Array<IVoteInfo> {
+    return this.allVotes.filter(voteInfo => voteInfo.vote === true || voteInfo.vote === false);
+  }
+
+  representativeVote(representativeAddress: string): boolean | null {
+    return this.votesArrayToMap(this.allVotes).get(representativeAddress);
+  }
+
+  representativeVoted(representativeAddress: string): boolean {
```

Comment:
```suggestion
  hasRepresentativeVoted(representativeAddress: string): boolean {
```
because only one letter diff with `representativeVote`, so hard to spot